File: Data Structures & Algorithms/asteroid-collision/submission-2.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = []

        for c in asteroids:
            while stack and c < 0 and stack[-1] > 0:
                if abs(c) == stack[-1]:
                    stack.pop()
                    break
                elif abs(c) > stack[-1]:
                    stack.pop()
                    continue
                else:
                    break
            else:
                stack.append(c)
        
        return stack

File: Data Structures & Algorithms/simplify-path/submission-2.rs
impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut stack: Vec<&str> = Vec::new();

        for component in path.split('/') {
            match component {
                "." | "" => continue,
                ".." => {
                    stack.pop();
                }
                _ => {
                    stack.push(component);
                }
            }
        }
        String::from("/") + &stack.join("/") 
    }
}
